File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/friend/FriendSpouseOnlineInformations.scala





















// Generated on 05/03/2013 10:35:06
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait FriendSpouseOnlineInformations extends FriendSpouseInformations {
  override val typeId = 93

  protected trait Type extends FriendSpouseInformations.Type with Serializable {
    override val typeId = FriendSpouseOnlineInformations.typeId

    val inFight: Boolean
    val followSpouse: Boolean
    val pvpEnabled: Boolean
    val mapId: Int
    val subAreaId: Short

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      var flag1 = 0
      flag1 = BooleanByteWrapper.setFlag(flag1, 0, inFight)
      flag1 = BooleanByteWrapper.setFlag(flag1, 1, followSpouse)
      flag1 = BooleanByteWrapper.setFlag(flag1, 2, pvpEnabled)
      buf.writeUByte(flag1.toShort)
      buf.writeInt(mapId)
      buf.writeShort(subAreaId)
    }

  }

  protected def create(spouseAccountId: Int, spouseId: Int, spouseName: String, spouseLevel: UByte, breed: Byte, sex: Byte, spouseEntityLook: EntityLook.TypeImpl, guildInfo: BasicGuildInformations.TypeImpl, alignmentSide: Byte, inFight: Boolean, followSpouse: Boolean, pvpEnabled: Boolean, mapId: Int, subAreaId: Short): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val spouseAccountId = buf.readInt()
    val spouseId = buf.readInt()
    val spouseName = buf.readString()
    val spouseLevel = buf.readUByte()
    val breed = buf.readByte()
    val sex = buf.readByte()
    val spouseEntityLook = EntityLook.deserialize(buf)
    val guildInfo = BasicGuildInformations.deserialize(buf)
    val alignmentSide = buf.readByte()
    val flag1 = buf.readUByte()
    val inFight = BooleanByteWrapper.getFlag(flag1, 0)
    val followSpouse = BooleanByteWrapper.getFlag(flag1, 1)
    val pvpEnabled = BooleanByteWrapper.getFlag(flag1, 2)
    val mapId = buf.readInt()
    val subAreaId = buf.readShort()

    create(spouseAccountId, spouseId, spouseName, spouseLevel, breed, sex, spouseEntityLook, guildInfo, alignmentSide, inFight, followSpouse, pvpEnabled, mapId, subAreaId)
  }

}

object FriendSpouseOnlineInformations extends FriendSpouseOnlineInformations {
  case class TypeImpl(spouseAccountId: Int, spouseId: Int, spouseName: String, spouseLevel: UByte, breed: Byte, sex: Byte, spouseEntityLook: EntityLook.TypeImpl, guildInfo: BasicGuildInformations.TypeImpl, alignmentSide: Byte, inFight: Boolean, followSpouse: Boolean, pvpEnabled: Boolean, mapId: Int, subAreaId: Short) extends Type
  override def create(spouseAccountId: Int, spouseId: Int, spouseName: String, spouseLevel: UByte, breed: Byte, sex: Byte, spouseEntityLook: EntityLook.TypeImpl, guildInfo: BasicGuildInformations.TypeImpl, alignmentSide: Byte, inFight: Boolean, followSpouse: Boolean, pvpEnabled: Boolean, mapId: Int, subAreaId: Short) = TypeImpl(spouseAccountId, spouseId, spouseName, spouseLevel, breed, sex, spouseEntityLook, guildInfo, alignmentSide, inFight, followSpouse, pvpEnabled, mapId, subAreaId)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(spouseAccountId: Int, spouseId: Int, spouseName: String, spouseLevel: UByte, breed: Byte, sex: Byte, spouseEntityLook: EntityLook.TypeImpl, guildInfo: BasicGuildInformations.TypeImpl, alignmentSide: Byte, inFight: Boolean, followSpouse: Boolean, pvpEnabled: Boolean, mapId: Int, subAreaId: Short) = create(spouseAccountId, spouseId, spouseName, spouseLevel, breed, sex, spouseEntityLook, guildInfo, alignmentSide, inFight, followSpouse, pvpEnabled, mapId, subAreaId)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/FriendSpouseOnlineInformations.java


// Generated on 05/08/2013 19:38:02
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class FriendSpouseOnlineInformations extends FriendSpouseInformations {
    public static final short TYPE_ID = 93;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public boolean inFight;
    public boolean followSpouse;
    public boolean pvpEnabled;
    public int mapId;
    public short subAreaId;
    
    public FriendSpouseOnlineInformations() { }
    
    public FriendSpouseOnlineInformations(int spouseAccountId, int spouseId, String spouseName, short spouseLevel, byte breed, byte sex, EntityLook spouseEntityLook, BasicGuildInformations guildInfo, byte alignmentSide, boolean inFight, boolean followSpouse, boolean pvpEnabled, int mapId, short subAreaId) {
        super(spouseAccountId, spouseId, spouseName, spouseLevel, breed, sex, spouseEntityLook, guildInfo, alignmentSide);
        this.inFight = inFight;
        this.followSpouse = followSpouse;
        this.pvpEnabled = pvpEnabled;
        this.mapId = mapId;
        this.subAreaId = subAreaId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        short flag1 = 0;
        flag1 = BooleanByteWrapper.setFlag(flag1, 0, inFight);
        flag1 = BooleanByteWrapper.setFlag(flag1, 1, followSpouse);
        flag1 = BooleanByteWrapper.setFlag(flag1, 2, pvpEnabled);
        buf.writeUByte(flag1);
        buf.writeInt(mapId);
        buf.writeShort(subAreaId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        short flag1 = buf.readUByte();
        inFight = BooleanByteWrapper.getFlag(flag1, 0);
        followSpouse = BooleanByteWrapper.getFlag(flag1, 1);
        pvpEnabled = BooleanByteWrapper.getFlag(flag1, 2);
        mapId = buf.readInt();
        if (mapId < 0)
            throw new RuntimeException("Forbidden value on mapId = " + mapId + ", it doesn't respect the following condition : mapId < 0");
        subAreaId = buf.readShort();
        if (subAreaId < 0)
            throw new RuntimeException("Forbidden value on subAreaId = " + subAreaId + ", it doesn't respect the following condition : subAreaId < 0");
    }
    
}
